### PR TITLE
Various fixes/improvements

### LINKED ISSUE #594: Trade page selection - truncate precision
On the trade page, when you list your tokens in the from/to, we use the full ETH / BTC precision. It would be ideal to truncate these, to e.g. 9 decimal points. 

Would this be blocked on a `display decimals` field in assetlists?

### LINKED ISSUE #623: `@osmosis-labs/math`: remove static name-spacing classes
### Problem
We organize math functions into a single file with static classes for name-spacing and code organization. This is unconventional and may cause bloat problems when bundling.

### Proposed Solution
Idiomatic java(type)script uses files as a name-spacing mechanism for organizing code. We should replace the current namespace class structure in favor of a file tree for code organization.

### LINKED ISSUE #624: Mobile trade page: token select modal
### Problem
The token select dropdowns on mobile can be hard to use, especially on smaller devices.

### Proposed Solution
As we did with the sort menu, use a modal to select tokens instead of a dropdown.

### LINKED ISSUE #630: Trade clipboard wallet status & slippage calculations
### Problem
As @syed8rahman pointed out. There are some problems with new trade clipboard that were not present in [old frontend](https://testnet.osmosis.zone/?from=ATOM&to=OSMO):
* All estimate calculations give up (go to 0) after > 10% slippage
* If a user hasn't connected a wallet, the connect wallet button shouldn't turn red on high slippage

example

### Proposed Solution

* Wallet disconnected
  - [x]  Don't show insufficient balance errors
  - [x] Don't turn button red
* Calculations
   - [ ] Don't zero out on high slippage

Currently, the red UI is based on an arbitrary/hard-coded slippage threshold, should we instead look at the user's slippage setting to determine what is a high slippage?

### LINKED ISSUE #631: Pools page: switch to All Pools list when searching
### Problem
CleanShot 2022-06-17 at 12 24 03@2x

When searching, it can be misleading/annoying to not get results since you haven't switched to the All Pools page.

### Proposed Solution
We should auto-switch to the All Pools table when users get "unsatisfactory" search results. this requires us to define what unsatisfactory results are, perhaps we can look at the score given by fuse.js for each result.

Also, we may want to indicate to the user that we've switched to the "All Pools" table, perhaps by flashing the control as it automatically switches as they are searching.

### LINKED ISSUE #635: Add preview image
Add a preview image so that, in cases like sharing on iMessage, a vibrant image would appear. Would help with credibility when it is shared/linked.

Screen Shot 2022-06-18 at 2 46 58 PM
 
I believe it is as simple as making an image available at a certain endpoint.

### LINKED ISSUE #640: Landing page loads random tokens depending on web browser
Uppon visiting the website for the first time on a new browser. The UI loads different tokens pairs on the default page depending on the browser.

Step to reproduce, images and video from internal QA testing:
https://github.com/osmosis-labs/test-management/issues/43

### PR DESCRIPTION
**Pull request by jonator**
* Refine banners
* Add OGP meta preview content to show title/image when sharing app link on iMessage, Twitter, etc. (Closes #635)

`@osmosis-labs/math`
* JS-module idiomatic rearrangement, remove static name-spacing classses - new major version publish on merge (Closes #623)

### Trade Page
* Mobile - token select modal instead of dropdown (Closes #624)
* Improve reliability of token select query params for first-time browsers (Closes #640)
* Don't display any error UI on trade clipboard when wallet not connected (Closes #630)
* Precision and to denom improvement (Closes #594 )

### Pools Page
* Auto expand searchable pool set as you type in the search bar (Closes #631)
  * All pools, then low TVL pools
  * @victortokaito look into dissolve prototype to clearly signal that the tabs switching is intentional (not a bug)
  * Returns these settings to how they we're before auto switching

### CODE REVIEW

**Comment by jonator** [comment]
Going to add cosmodal

**Comment by jonator** [comment]
Actually nevermind, I saw work is pending on cosmodal package to add dao dao's improvements

**Comment by victortokaito** [comment]
https://www.figma.com/file/wQjMyxY0EnEk29gBzGDMe5?node-id=4609:2451#209550950

**Comment by victortokaito** [comment]
@jonator [search auto-switches to 'all pools':  prototype uses dissolve animation to clearly signal that the tabs switching is intentional (not a bug)
](
https://www.figma.com/proto/7iuO8MZeiYezTnF7suwTNN/draft-UI?page-id=1025%3A41711&node-id=1027%3A49232&viewport=623%2C497%2C0.25&scaling=min-zoom&starting-point-node-id=1027%3A49232)

File: packages/web/modals/ibc-transfer.tsx [hunk 1]
```diff
@@ -70,7 +68,12 @@ export const IbcTransferModal: FunctionComponent<ModalBaseProps & IbcTransfer> =
         props.onRequestClose
       );
 
-    // Mobile only - copy to clipboard
+    console.log(
```
Comment (delivan) [inline packages/web/modals/ibc-transfer.tsx:71]: Remove console log

**Review by delivan — APPROVED** [review]

File: packages/web/modals/ibc-transfer.tsx (hunk 1 above)
Comment (delivan) [reply packages/web/modals/ibc-transfer.tsx:71]: except it, LGTM🚀

**Pull request merged**